openml-utils/src/test/java/com/feedzai/util/provider/AbstractProviderModelBaseTest.java

```diff
@@ -356,6 +403,15 @@ protected L getMachineLearningModelLoader(final MLAlgorithmEnum algorithm) {
      */
     public abstract Instance getDummyInstance();
 
+    /**
+     * Gets a dummy {@link Instance} to be classified by a {@link ClassificationMLModel} that returns different results
+     * from the results produced by {@link #getDummyInstance()};
+     *
+     * @return a dummy {@link Instance}.
+     * @since 0.2.0
+     */
+    public abstract Instance getDummyInstanceDifferentResult();
```

Comment:
I have a doubt: why can't we implement the `getDummyInstance` and `getDummyInstanceDifferentResult` methods in this class, and allow providers to override if needed? I guess some providers have some constraints on the instance schema (scikit?), but most should be able to deal with a normal instance no?

either way, that's probably a too big of a change for now

Reply:
each of the provider needs their own implementation of these methods.
Their models were not trained with the same data and so if we tried to classify "generic instances" it would fail.
On the other hand we could create new models for a common dataset schema, except for python providers, but we would need to refactor all UTs that verify providers.